### PR TITLE
Greg/privatemagic

### PR DESCRIPTION
**Pull request by GodGreg**

### CODE REVIEW

File: src/Components/Admin/UserManagement.jsx [hunk 1]
```diff
@@ -55,8 +56,14 @@ function convertToTableData(fetched, programs) {
...
-function UserManagement() {
-  const [loadUsers, reloadUsers] = usePromise(GET.getAllUsersAPI, {}, []);
+function UserManagement({ program }) {
+  const [loadUsers, reloadUsers] = usePromise(
+    GET.getAllProgramUsers,
+    { program },
+    [program]
+  );
+
+  console.log(loadUsers);
```
Comment (ahmedhamodi) [inline src/Components/Admin/UserManagement.jsx:66]: Console log

File: src/Components/Admin/UserManagement.jsx [hunk 2]
```diff
@@ -68,6 +75,8 @@ function UserManagement() {
     [loadUsers, programs]
   );
 
+  console.log("We're at user Managment");
```
Comment (ahmedhamodi) [inline src/Components/Admin/UserManagement.jsx:78]: Yes you are!

File: src/Components/Header/Header.jsx [hunk 3]
```diff
@@ -122,25 +122,22 @@ function Header({ program, loadProgram, history, admin, curRoute, routes }) {
...
-  }
+  const hasAdminAccessForCurrentProgram = true;
+  // if (programsMap[program]) {
+  //   const userProgram = appUser.programs.find((prog) => {
+  //     return programsMap[program]._id === prog.id;
+  //   });
+  //   if (userProgram && userProgram.role === "admin") {
+  //     hasAdminAccessForCurrentProgram = true;
+  //   }
+  // }
```
Comment (ahmedhamodi) [inline src/Components/Header/Header.jsx:140]: Perhaps leave a comment or a TODO is there's future action to be done on this. Otherwise, remove.

**Review by ahmedhamodi — COMMENTED** [review]
Small changes!

File: src/Components/Admin/UserManagement.jsx (hunk 2 above)
Comment (GodGreg) [reply src/Components/Admin/UserManagement.jsx]: shhhhhhh hahaha

File: src/Components/Header/Header.jsx (hunk 3 above)
Comment (GodGreg) [reply src/Components/Header/Header.jsx:141]: I added comment

File: src/Authentication/PrivateRoute.js [hunk 4]
```diff
@@ -1,52 +1,107 @@
...
-      show
-      spinnerProps={{
-        radius: 220,
-        stroke: 2
-      }}
-    />
-  ) : roleAccess ? (
+    //If we reach here then there exists exactly one element in userPrograms which is the users current program
+    let userCurrentProgram = userPrograms.value.filter(
+      (userProgram) => userProgram.id === program
```
Comment (GodGreg) [inline src/Authentication/PrivateRoute.js:64]: Could have used "find" smh greg

File: src/Components/Header/Header.jsx [hunk 5]
```diff
@@ -194,48 +192,55 @@ function Header({ program, loadProgram, history, admin, curRoute, routes }) {
...
-                            visible={(
-                              route.title !== curRoute.title
-                            ).toString()}
-                          >
-                            {route.title}
+              <p style={{ marginLeft: "4px", margin: "12px" }}>
+                {" "}
+                {curRoute.title}{" "}
+              </p>
+              {routes.length > 0 ? (
```
Comment (GodGreg) [inline src/Components/Header/Header.jsx:199]: These changes are for displaying the dropdown or not if there are no routes to show.

**Review by brittanylau — APPROVED** [review]
Approving for functionality, trusting @ahmedhamodi's review of the code itself :)

**Pull request merged**